sdk/src/stream_config/config_iggy_consumer.rs

```diff
@@ -0,0 +1,369 @@
+            polling_strategy,
+            partitions_count,
+            replication_factor,
+            encryptor,
+        }
+    }
+
+    /// Creates a new `IggyConsumerConfig` from the given arguments.
+    ///
+    /// # Args
+    ///
+    /// * `stream` - The stream name.
+    /// * `topic` - The topic name.
+    /// * `batch_size` - The max number of messages to send in a batch.
+    /// * `polling_interval` - The interval between polling for new messages.
+    ///
+    /// Returns:
+    /// A new `IggyConsumerConfig`.
+    ///
+    pub fn from_stream_topic(
+        stream: &str,
+        topic: &str,
+        batch_size: u32,
+        polling_interval: IggyDuration,
+    ) -> Self {
+        let stream_id = shared_config::get_identifier_from_string(stream);
+        let topic_id = shared_config::get_identifier_from_string(topic);
+
+        Self {
+            stream_id,
+            stream_name: stream.to_string(),
+            topic_id,
+            topic_name: topic.to_string(),
+            auto_commit: AutoCommit::When(AutoCommitWhen::PollingMessages),
+            batch_size,
+            create_stream_if_not_exists: false,
+            create_topic_if_not_exists: false,
+            consumer_name: format!("consumer-{}-{}", stream, topic),
+            consumer_kind: ConsumerKind::ConsumerGroup,
+            polling_interval,
+            polling_strategy: PollingStrategy::last(),
+            partitions_count: 1,
+            replication_factor: None,
+            encryptor: None,
+        }
+    }
+}
+
+impl IggyConsumerConfig {
+    pub fn stream_id(&self) -> &Identifier {
+        &self.stream_id
+    }
+
+    pub fn stream_name(&self) -> &str {
+        &self.stream_name
+    }
+
+    pub fn topic_id(&self) -> &Identifier {
+        &self.topic_id
+    }
+
+    pub fn topic_name(&self) -> &str {
+        &self.topic_name
+    }
+
+    pub fn auto_commit(&self) -> AutoCommit {
+        self.auto_commit
+    }
+
+    pub fn batch_size(&self) -> u32 {
+        self.batch_size
+    }
+    pub fn create_stream_if_not_exists(&self) -> bool {
+        self.create_stream_if_not_exists
+    }
+
+    pub fn create_topic_if_not_exists(&self) -> bool {
+        self.create_topic_if_not_exists
+    }
+
+    pub fn consumer_name(&self) -> &str {
+        &self.consumer_name
+    }
+
+    pub fn consumer_kind(&self) -> ConsumerKind {
+        self.consumer_kind
+    }
+
+    pub fn polling_interval(&self) -> IggyDuration {
+        self.polling_interval
+    }
+
+    pub fn polling_strategy(&self) -> PollingStrategy {
+        self.polling_strategy
+    }
+
+    pub fn partitions_count(&self) -> u32 {
+        self.partitions_count
+    }
+
+    pub fn replication_factor(&self) -> Option<u8> {
+        self.replication_factor
+    }
+
+    pub fn encryptor(&self) -> &Option<Arc<EncryptorKind>> {
+        &self.encryptor
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_builder() {
```

Comment:
Tests look good! You can also check the naming convention we use in most places, a bit similar to BDD testing approach e.g. `given`, `should` etc. Anyway, I think for such test cases, either is fine :)